cincinnati/src/plugins/internal/cincinnati_graph_fetch.rs

```diff
@@ -0,0 +1,281 @@
+//! This plugin implements fetching a Cincinnati graph via HTTP from a `/v1/graph`-compliant endpoint.
+
+extern crate futures;
+extern crate quay;
+extern crate tokio;
+
+use crate::plugins::{InternalIO, InternalPlugin, InternalPluginWrapper, PluginSettings};
+use failure::Fallible;
+use plugins::AsyncIO;
+use plugins::PluginReference;
+use prometheus::{Counter, Registry};
+use std::fmt::{Debug, Formatter};
+
+/// Default URL to upstream graph provider.
+pub static DEFAULT_UPSTREAM_URL: &str = "http://localhost:8080/v1/graph";
+
+/// Plugin settings.
+#[derive(Clone, Debug, Deserialize, SmartDefault)]
+#[serde(default)]
+struct CincinnatiGraphFetchSettings {
+    #[default(DEFAULT_UPSTREAM_URL.to_string())]
+    upstream: String,
+}
+
+/// Graph fetcher for Cincinnati `/v1/graph` endpoints.
+pub struct CincinnatiGraphFetchPlugin {
+    pub upstream: String,
+    pub prometheus_registry: Option<Registry>,
+    pub http_upstream_reqs: Option<Counter>,
+    pub http_upstream_unreachable: Option<Counter>,
```

Comment:
Why are these two `Option`al?

Reply:
Same as for the registry, I wanted to make the metrics optional.